_includes/pbc/all/install-features/202307.0/install-the-dynamic-store-prices-feature.md

```diff
@@ -0,0 +1,517 @@
+        return [
+            ...
+            new CurrencyBackendGatewayApplicationPlugin(),
+            ...
+        ];
+    }
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure service container has `currency` service.
+
+{% endinfo_block %}
+
+**src/Pyz/Zed/Store/StoreDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\Store;
+
+use Spryker\Zed\Currency\Communication\Plugin\Store\CurrencyStoreCollectionExpanderPlugin;
+use Spryker\Zed\Currency\Communication\Plugin\Store\CurrencyStorePostCreatePlugin;
+use Spryker\Zed\Currency\Communication\Plugin\Store\CurrencyStorePostUpdatePlugin;
+use Spryker\Zed\Currency\Communication\Plugin\Store\DefaultCurrencyStorePreCreateValidationPlugin;
+use Spryker\Zed\Currency\Communication\Plugin\Store\DefaultCurrencyStorePreUpdateValidationPlugin;
+use Spryker\Zed\Store\StoreDependencyProvider as SprykerStoreDependencyProvider;
+
+class StoreDependencyProvider extends SprykerStoreDependencyProvider
+{
+    /**
+     * @return array<\Spryker\Zed\StoreExtension\Dependency\Plugin\StorePreCreateValidationPluginInterface>
+     */
+    protected function getStorePreCreateValidationPlugins(): array
+    {
+        return [ 
+            ...
+            new DefaultCurrencyStorePreCreateValidationPlugin(),
+            ...
+        ];
+    }
+
+    /**
+     * @return array<\Spryker\Zed\StoreExtension\Dependency\Plugin\StorePreUpdateValidationPluginInterface>
+     */
+    protected function getStorePreUpdateValidationPlugins(): array
+    {
+        return [ 
+            ...
+            new DefaultCurrencyStorePreUpdateValidationPlugin(),
+            ...
+        ];
+    }
+
+    /**
+     * @return array<\Spryker\Zed\StoreExtension\Dependency\Plugin\StorePostCreatePluginInterface>
+     */
+    protected function getStorePostCreatePlugins(): array
+    {
+        return [
+            ...
+            new CurrencyStorePostCreatePlugin(),
+            ...
+        ];
+    }
+
+    /**
+     * @return array<\Spryker\Zed\StoreExtension\Dependency\Plugin\StorePostUpdatePluginInterface>
+     */
+    protected function getStorePostUpdatePlugins(): array
+    {
+        return [
+            ...
+            new CurrencyStorePostUpdatePlugin(),
+            ...
+        ];
+    }
+
+    /**
+     * @return array<\Spryker\Zed\StoreExtension\Dependency\Plugin\StoreCollectionExpanderPluginInterface>
+     */
+    protected function getStoreCollectionExpanderPlugins(): array
+    {
+        return [
+            ...
+            new CurrencyStoreCollectionExpanderPlugin(),
+            ...
+        ];
+    }
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+Steps: 
+- Make sure that you get an error message if you try to create or update a store with a default currency that is not assigned to the store.
+- Make sure that `spy_currency_store` table has been updated with the default currency for the created store or updated store.
+- Make sure expanded store transfers have currency codes.
```

Comment:
Why and when should this happen?

Reply:
It should be done in the code by checking response transfer on creation, updating or getting all stores actions